### PR TITLE
fix(nuxt): stop loading indicator on navigation failure

### LINKED ISSUE #21750: `<NuxtLoadingIndicator>` gets stuck on `abortNavigation` from middleware
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v16.20.0
- Nuxt Version:     3.6.0
- Nitro Version:    2.5.1
- Package Manager:  npm@9.4.2
- Builder:          vite
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -
------------------------------

But it happens in any OS too.

### Reproduction

https://stackblitz.com/edit/github-fwqsly?file=app.vue

### Describe the bug

When you use the router to navigate to a page that has some middleware, if it returns `abortNavigation()` for some reason the `<NuxtLoadingIndicator>` component glitches and doesn't disappear.

I expect it to complete the loading or not show because of the `throttle` property, but it shows the loading bar with very glitchy behavior and it never disappears.

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by huang-julien**
### 🔗 Linked issue
fix https://github.com/nuxt/nuxt/issues/21750

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
Hi :wave: this PR fixes #21750 on any navigation failure and when an router `onError` occurs

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### todo 
- [x] test fixtures

### CODE REVIEW

**Comment by danielroe** [comment]
Is it sufficient to use router.onError to solve the linked issue instead of adding a new hook?

**Comment by huang-julien** [comment]
oh nice point, I think we need both. 

When using `abortNavigation()` without any argument, it returns false and doesn't throw so doesn't goes into the `router.onError` hook. 

But when having an error argument, it does not goes within `app:navigation:aborted` hook because it returns false instead of throwing.

**Comment by huang-julien** [comment]
hmmm... that's the kind of PR i'd like to test with nuxt-vitest 😅

**Review by danielroe — APPROVED** [review]

File: packages/nuxt/src/app/components/nuxt-loading-indicator.ts [hunk 1]
```diff
@@ -37,13 +38,24 @@ export default defineComponent({
...
+      indicator.finish()
+    })
     router.beforeResolve((to, from) => {
       if (to === from || to.matched.every((comp, index) => comp.components && comp.components?.default === from.matched[index]?.components?.default)) {
         indicator.finish()
       }
     })
+
+    router.afterEach((_to, _from, failure) => {
+      if (isNavigationFailure(failure, NavigationFailureType.aborted)) {
```
[the remark this answers is not in the document]
Comment (danielroe) [reply packages/nuxt/src/app/components/nuxt-loading-indicator.ts:51]: Yes, probably! I don't think we want to continue loading in any case.

**Pull request merged**